url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/StandAloneSimulation.java

```diff
@@ -0,0 +1,527 @@
+/*   
+   Copyright 2011-2013 The Cassandra Consortium (cassandra-fp7.eu)
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package eu.cassandra.sim;
+
+import java.io.File;
+import java.util.Date;
+import java.util.Hashtable;
+import java.util.Properties;
+import java.util.TreeMap;
+import java.util.Vector;
+
+import eu.cassandra.sim.entities.appliances.Appliance;
+import eu.cassandra.sim.entities.appliances.ConsumptionModel;
+import eu.cassandra.sim.entities.installations.Installation;
+import eu.cassandra.sim.entities.people.Activity;
+import eu.cassandra.sim.entities.people.Person;
+import eu.cassandra.sim.math.Gaussian;
+import eu.cassandra.sim.math.GaussianMixtureModels;
+import eu.cassandra.sim.math.Histogram;
+import eu.cassandra.sim.math.ProbabilityDistribution;
+import eu.cassandra.sim.math.Uniform;
+import eu.cassandra.sim.utilities.SetupFileParser;
+
+/**
+ * The StandAloneSimulation class includes an implementation of the setupScenario() method that
+ * parses a properly formatted input text file and  <br>
+ * (a) sets up the installations (or installation types) to be included in the simulation, <br>
+ * (b) sets up the pricing scheme(s) to be used in the simulation, <br>
+ * (c) defines the set of simulation parameters to be used and, in case of dynamic scenarios, <br>
+ * (d) the demographic data, according to which the entities involved in the simulation will be instantiated.
+ * 
+ * @author Fani A. Tzima (fani [dot] tzima [at] iti [dot] gr)
+ * 
+ */
+public class StandAloneSimulation extends Simulation {
+	
+	/**
+	 * Instantiates a new stand alone simulation, whose scenario setup is stored in a properly formatted input file.
+	 *
+	 * @param outputPath the path to the output directory
+	 * @param dbName the name of the database where output data will be stored
+	 * @param seed the seed for the random number generator
+	 * @param useDerby whether to use MongoDB (false) or Apache Derby (true) for storing simulation output data
+	 */
+	public StandAloneSimulation(String outputPath, String dbName, int seed, boolean useDerby) {
+		super(outputPath, dbName, seed, useDerby);
+	}
+	
+	
+	/**
+	 * Specific implementation of the  {@link eu.cassandra.sim.Simulation#setupScenario() setupScenario} method
+	 * that parses the information required to setup a scenario from a properly formatted input file.
+	 */
+	@SuppressWarnings("deprecation")
+	@Override
+	public Vector<Installation> setupScenario()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 52. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 86-110 into a separate method.